src/partition/Partition.hpp

```diff
@@ -45,14 +45,14 @@ class Partition {
   /// The partition is computed, i.e. the mesh re-partitioned if required and all data structures are set up.
   virtual void compute() = 0;
 
-  void setFromMapping(mapping::PtrMapping fromMapping)
+  void addFromMapping(mapping::PtrMapping fromMapping)
   {
-    _fromMapping = fromMapping;
+    _fromMappings.push_back(fromMapping);
```

Comment:
We can move the argument to prevent yet another unnecessary copy of the shared ptr.

```suggestion
    _fromMappings.push_back(std::move(fromMapping));
```